dimod/include/dimod/adjvectorbqm.h

```diff
@@ -209,9 +209,19 @@ class AdjVectorBQM {
             adj.resize(max_label + 1);
         }
 
-        // we can get 5-10% speedup on dense problems by counting the degrees
-        // and using that to reserve the neighborhood vectors. However, since
-        // it uses more memory and slows down sparse problems, we don't.
+        // Count the degrees and use that to reserve the neighborhood vectors
```

Comment:
Turns out the memory saved is valuable as well (memory savings come from not over-allocating the vectors).